File: tests/messenger_test.py
from py.yeti import Messenger
import time

m0 = Messenger("m0")
m1 = Messenger("m1")
m2 = Messenger("m2")
m3 = Messenger("m3")
m4 = Messenger("m4")
m5 = Messenger("m5")


def on_m0_telegram(data):
    print(data["msg"])
    return {"msg": "Gotcha!"}

m0.register_message_type("telegram", on_m0_telegram)

m4.register_messenger_address("m0", "127.0.0.1", 9000, 9010)
m2.register_messenger_address("m4", "127.0.0.1", 9004, 9014)
m5.register_messenger_address("m2", "127.0.0.1", 9002, 9012)
m2.register_messenger_address("m5", "127.0.0.1", 9005, 9015)
m5.register_messenger_address("m4", "127.0.0.1", 9004, 9014)
m0.register_messenger_address("jm2", "127.0.0.1", 9103, 9104)

m0.start_server("127.0.0.1", 9000, 9010)
m1.start_server("127.0.0.1", 9001, 9011)
m2.start_server("127.0.0.1", 9002, 9012)
m3.start_server("127.0.0.1", 9003, 9013)
m4.start_server("127.0.0.1", 9004, 9014)
m5.start_server("127.0.0.1", 9005, 9015)

response = m5.send_message("telegram", {"msg": "Hi!"}, "jm0", blocking=True)
print(response)
time.sleep(10000)

File: java/src/yeti/MessengerTest.java
package yeti;

import static org.junit.Assert.*;

import java.io.IOException;

import org.json.JSONException;
import org.json.JSONObject;
import org.junit.Test;

import yeti.Messenger.CouldNotResolveMessengerException;

public class MessengerTest {
	
	public JSONObject onMessage(JSONObject message){
		System.out.println("Bingo!");
		System.out.println(message.getString("msg"));
		return new JSONObject("{\"msg\": \"Awesomeness!\";}");
	}

	@Test
	public void test() throws IOException, JSONException, CouldNotResolveMessengerException, InterruptedException {
		Messenger m0 = new Messenger("jm0");
		Messenger m1 = new Messenger("jm1");
		Messenger m2 = new Messenger("jm2");
		
		m0.registerMessageType("telegram", this::onMessage);
		m0.registerMessengerAddress("m5", "127.0.0.1", 9005, 9015);
		m1.registerMessengerAddress("jm0", "127.0.0.1", 9100, 9101);
		m2.registerMessengerAddress("jm1", "127.0.0.1", 9102, 9103);
		
		m0.startServer("127.0.0.1", 9100, 9101);
		m1.startServer("127.0.0.1", 9102, 9103);
		m2.startServer("127.0.0.1", 9103, 9104);
		Thread.sleep(1000);
		//JSONObject response1 = m2.sendMessage("telegram", new JSONObject("{\"msg\": \"String from the land of Java!\";}"), "m0", true);
		//System.out.println(response1.toString());
		//JSONObject response2 = m2.sendMessage("telegram", new JSONObject("{\"msg\": \"String!\";}"), "m0", true);
		//System.out.println(response2.toString());	
		
		Thread.sleep(10000);
		
		m0.stopServer();
		m1.stopServer();
		m2.stopServer();
	}

}
